runtime/gc_base/MixedObjectAllocationModel.hpp

```diff
@@ -122,6 +125,9 @@ class MM_MixedObjectAllocationModel : public MM_JavaObjectAllocationModel
 		if (getAllocateDescription()->getPreHashFlag()) {
 			env->getExtensions()->objectModel.initializeHashSlot((J9JavaVM*)env->getLanguageVM(), objectPtr);
 		}
+		if (J9_ARE_ANY_BITS_SET(_class->classFlags, J9ClassEnsureHashed)) {
+			VM_ObjectHash::setHasBeenHashed((J9JavaVM *)env->getLanguageVM(), objectPtr);
```

Comment:
When I referred to this helper I thought about calling from VM side. In Object Model however please create new helper similar to existing using `setObjectFlags()` (Object Model handling for Mixed build)

Reply:
An example may be `setObjectHasBeenMoved()` - this function set both flags and you need only one - call it`setObjectHasBeenHashed()`. 
Also `VM_ObjectHash::setHasBeenHashed()` set flags atomically. I believe there is need to do it in this case.

Reply:
Since we're setting `OMR_GC_ALLOCATE_OBJECT_HASHED` in mgcalloc, I assume this change is not necessary anymore? We just check that it is in the flags and initialize the slot.

Reply:
yes, I think so